docs/src/howto/deploy/gcp.md

```diff
@@ -137,6 +137,175 @@ When you first open the lakeFS UI, you will be asked to create an initial admin
+
+# Data service account (accesses everything except _lakefs/**, network-restricted)
+gcloud iam service-accounts create "$SA_RESTRICTED_ID" \
+  --project="$PROJECT_ID" \
+  --description="lakeFS data - RW everywhere except /_lakefs/** (network-restricted)" \
+  --display-name="lakeFS Data Service Account"
+```
+
+Get the full service account emails:
+
+```bash
+export SA_OPEN="$SA_OPEN_ID@$PROJECT_ID.iam.gserviceaccount.com"
+export SA_RESTRICTED="$SA_RESTRICTED_ID@$PROJECT_ID.iam.gserviceaccount.com"
+```
+
+#### 2. Configure Bucket IAM Policies
+
+Grant the metadata service account access to `_lakefs/**` prefix only:
+
+```bash
+gcloud storage buckets add-iam-policy-binding "gs://$BUCKET" \
+  --member="serviceAccount:$SA_OPEN" \
+  --role="roles/storage.objectAdmin" \
+  --condition='title=lakefs-metadata-only,description=Access_only_to_lakefs_prefix,expression=resource.type == "storage.googleapis.com/Object" && resource.name.extract("/objects/{before}/_lakefs/") != ""'
+```
+
+Grant the data service account access to everything except `_lakefs/**`:
+
+```bash
+# Object access (read/write) for non-_lakefs paths
+gcloud storage buckets add-iam-policy-binding "gs://$BUCKET" \
+  --member="serviceAccount:$SA_RESTRICTED" \
+  --role="roles/storage.objectAdmin" \
+  --condition='title=lakefs-data-only,description=Access_except_lakefs_prefix,expression=resource.type == "storage.googleapis.com/Object" && resource.name.extract("/objects/{before}/_lakefs/") == ""'
+```
+
+!!! warning "storage.objects.list Permission Required"
+    The data service account also needs the `storage.objects.list` permission without conditionals to perform listing operations. Since standard GCP roles also include `storage.objects.get` (which would allow reading `_lakefs/**` data), you'll need to create a custom IAM role that includes only the `storage.objects.list` permission, then grant it to the data service account without conditions.
+
+#### 3. Set Up VPC Service Controls
+
+Create an access level that defines your allowed networks (adjust IP ranges and VPC as needed):
+
+```bash
+# Get your organization's access policy ID
+export ORG_ID="[YOUR_ORG_ID]"
+export POLICY_ID=$(gcloud access-context-manager policies list \
+  --organization="$ORG_ID" \
+  --format="value(name)")
+
+# Create an access level for your allowed networks
+cat > access-level.yaml <<EOF
+- ipSubnetworks:
+  - "[YOUR_ALLOWED_IP_CIDR]"
+  vpcNetworkSources:
+  - vpcSubnetwork:
+      network: "projects/[YOUR_VPC_PROJECT]/global/networks/[YOUR_VPC_NAME]"
+EOF
+
+gcloud access-context-manager levels create Restrict_Network_Access \
+  --policy="$POLICY_ID" \
+  --title="Restrict Network Access to Approved IPs and VPCs" \
+  --combine-function=OR \
+  --basic-level-spec=access-level.yaml
+```
+
+#### 4. Create Ingress Policy
+
+Define the ingress policy for the restricted service account:
+
+```bash
+export ACCESS_LEVEL="accessPolicies/$POLICY_ID/accessLevels/Restrict_Network_Access"
```

Comment:
in this step we use bash variable expansion and  bash to create a file populated with values. in other steps we use placeholders.
consider using placeholders for this one too.